packages/hooks/use-fills.tsx

```diff
@@ -0,0 +1,71 @@
+import { useConnect, warn } from "frontity";
+import { Fill } from "@frontity/types";
+
+interface FillsPackage {
+  state: {
+    fills: { [key: string]: Fill };
+  };
+  libraries: {
+    fills: {
+      [key: string]: React.ComponentType;
+    };
+  };
+}
+
+const useFills = (name: string) => {
```

Comment:
This is missing TSDocs.

Maybe explicitly define the return type as well, so we cannot break it in a refactoring.

Reply:
👍   on the TSDoc.

The return type is:
<img width="381" alt="Screenshot 2020-05-22 at 14 42 44" src="https://user-images.githubusercontent.com/5417266/82704116-80227700-9c3a-11ea-8f0b-aa19a42f7d43.png">

because I assign a type here: https://github.com/frontity/frontity/blob/e2056a31a608e05d58f60c7ec98ee4ec5934209e/packages/hooks/use-fills.tsx#L47-L49

I need to specify the type of that accumulator so that I get type checks inside of the `.reduce` callback.

So the type inference figures out the correct return type of `useFills` for me. Is there a problem with this approach that I'm not aware of?

Reply:
Explicit return types make impossible to break the function (type-wise) in a subsequent refactoring.